### DISCUSSION TITLE
Question about securing?

### DISCUSSION THREAD

**Discussion by biapar** [directus/directus#9782] [Archive]
Hi,
I’ve a question about secure. 
To access to public API or some read/write API , I create a service user and its token.
How to secure the token or username and password for this service user client side ( vuejs application )? Thx

**Comment by rijkvanzanten** [comment]
There's no Directus-specific way of handling those tokens. If you want your users to login, I'd recommend using [the cookie mode of login](https://docs.directus.io/reference/authentication/#login) to ensure the client side JS can't read the refresh token, preventing XSS attacks on local variables. If you want to use a static token for your whole app, I would recommend adding some form of middleman server to proxy the requests to ensure the token doesn't leak
